components/brave_account/brave_account_service.cc

```diff
@@ -199,6 +181,50 @@ void BraveAccountService::CancelRegistration() {
   pref_service_->ClearPref(prefs::kBraveAccountVerificationToken);
 }
 
+void BraveAccountService::LoginInitialize(const std::string& email,
+                                          const std::string& serialized_ke1,
+                                          LoginInitializeCallback callback) {
+  if (email.empty() || serialized_ke1.empty()) {
+    return std::move(callback).Run(base::unexpected(mojom::LoginError::New()));
+  }
+
+  LoginInit::Request request;
+  request.network_traffic_annotation_tag =
+      net::MutableNetworkTrafficAnnotationTag(kTrafficAnnotation);
+  request.email = email;
+  request.serialized_ke1 = serialized_ke1;
+  Client<LoginInit>::Send(
+      url_loader_factory_, std::move(request),
+      base::BindOnce(&BraveAccountService::OnLoginInitialize,
+                     weak_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void BraveAccountService::LoginFinalize(
+    const std::string& encrypted_login_token,
+    const std::string& client_mac,
+    LoginFinalizeCallback callback) {
+  if (encrypted_login_token.empty() || client_mac.empty()) {
+    return std::move(callback).Run(base::unexpected(mojom::LoginError::New()));
+  }
+
+  const std::string login_token = Decrypt(encrypted_login_token);
+  if (login_token.empty()) {
+    return std::move(callback).Run(base::unexpected(mojom::LoginError::New(
+        std::nullopt, mojom::LoginErrorCode::kLoginTokenDecryptionFailed)));
+  }
+
+  WithHeaders<LoginFinalize::Request> request;
+  request.network_traffic_annotation_tag =
+      net::MutableNetworkTrafficAnnotationTag(kTrafficAnnotation);
+  request.client_mac = client_mac;
+  request.headers.SetHeader("Authorization",
+                            base::StrCat({"Bearer ", login_token}));
```

Comment:
At some point, it may be useful to create a helper function to set bearer tokens.

Reply:
Yeah, agreed. I'm thinking of adding a helper for this in the endpoint client in a follow-up.